code/modules/reagents/chems/chems_ethanol.dm

```diff
@@ -42,13 +42,25 @@
 	M.add_chemical_effect(CE_ALCOHOL_TOXIC, alcohol_toxicity)
 
 /decl/material/liquid/ethanol/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed, var/datum/reagents/holder)
-
 	..()
+
 	M.adjust_nutrition(nutriment_factor * removed)
 	M.adjust_hydration(hydration_factor * removed)
+
+	if(adj_temp > 0 && M.bodytemperature < targ_temp) // 310 is the normal bodytemp. 310.055
+		M.bodytemperature = min(targ_temp, M.bodytemperature + (adj_temp * TEMPERATURE_DAMAGE_COEFFICIENT))
+	if(adj_temp < 0 && M.bodytemperature > targ_temp)
+		M.bodytemperature = min(targ_temp, M.bodytemperature - (adj_temp * TEMPERATURE_DAMAGE_COEFFICIENT))
```

Comment:
The temperature shifting from ethanol should perhaps come after the metabolic check, as I assume the temperature shift is caused by expanding blood vessels?
What about creatures without blood (vessels)? Should perhaps even be moved to `affect_blood()` instead

Reply:
This should go after the metabolism check, I agree. Also the nutriment/hydration adjustment should be blocked by not being metabolized.